### PR TITLE
Add option to build without HDF5 parallel

### PR DESCRIPTION
**Pull request by jeanlucf22**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -53,12 +53,17 @@ else()
 endif()
 
 # HDF5 (required)
+set(MGMOL_USE_HDF5P TRUE CACHE BOOL "Use HDF5 parallel capability")
```
Comment (oseikuffuor1) [inline CMakeLists.txt:56]: @jeanlucf22 do we want this to be always true by default?

File: src/HDFrestart.cc [hunk 2]
```diff
@@ -499,6 +503,7 @@ HDFrestart::HDFrestart(const std::string& filename, const pb::PEenv& pes,
                 << "HDFrestart(): create file " << filename_ << std::endl;
         hid_t access_plist
             = H5Pcreate(H5P_FILE_ACCESS); // property list identifier
+#ifdef MGMOL_USE_HDF5P
```
Comment (oseikuffuor1) [inline src/HDFrestart.cc:508]: should similar changes to guard use_hdf5p_ be applied in HDFrestart.h as well? There are a few functions in there that use a this runtime check.

**Review by oseikuffuor1 — COMMENTED** [review]
@jeanlucf22 I left a couple comments for your review.

File: CMakeLists.txt (hunk 1 above)
Comment (jeanlucf22) [reply CMakeLists.txt:56]: that is making the option being what we have now, by default

Comment (oseikuffuor1) [reply CMakeLists.txt:56]: Oh ok. What about the other comment below? Should changes be applied to HDFrestart.h also?

Comment (jeanlucf22) [reply CMakeLists.txt:56]: yes, working on it

**Review by oseikuffuor1 — APPROVED** [review]
Looks good. Thanks @jeanlucf22.

**Pull request merged**